jupyterhub/templates/hub/service.yaml

```diff
@@ -5,6 +5,10 @@ metadata:
   labels:
     {{- include "jupyterhub.labels" . | nindent 4 }}
   annotations:
+    {{- if .Values.hub.service.annotatePrometheus }}
+    prometheus.io/scrape: "true"
+    prometheus.io/path: {{ .Values.hub.baseUrl }}hub/metrics
```

Comment:
Is there a way we can avoid the boolean flag? e.g. by checking for the keys in hub.service.annotations?

Reply:
Oh btw: I realize that this could fail if both are false now, then we will end up with a blank `annotations:`.

@minrk we cannot concatinate `.Values.hub.baseUrl` with hub/metrics within `values.yaml`. So, if we don't want a boolean flag but instead have it enabled no matter what without the option, we would need to merge the `hub.service.annotations` dict with a dynamically created dict containing the dynamic value of `prometheus.io/path: {{ .Values.hub.baseUrl }}hub/metrics`.

What makes you wish to avoid the boolean flag?

Reply:
I mean something like:

```yaml
prometheus.io/path: {{ default .Values.hub.service.annotations['prometheus.io/path'] .Values.hub.baseUrl + hub/metrics }}}
```

(that's not quite the right syntax, but I can't test it right now)

Reply:
@minrk I've got something working with the latest commit.

Reply:
Perfect, thanks!

Reply:
Nice solution!